tests/test_utils.sh

```diff
@@ -694,6 +699,63 @@ function test_path_format_last_dir() {
   assertEquals "full directory formatting with multichar separator" "{l}pathname" "$lp_path_format"
 }
 
+function test_path_links() {
+  typeset HOME="/home/user"
+  typeset PWD="/"
+  typeset SSH_CONNECTION="1.2.3.4 111 5.6.7.8 222"
+
+  _lp_find_vcs() {
+    return 1
+  }
+
+  LP_ENABLE_SHORTEN_PATH=0
+  LP_ENABLE_HYPERLINKS=0
+  LP_PATH_VCS_ROOT=1
+
+  typeset lp_path lp_path_format
+
+  _lp_path_format '{format}'
+  assertEquals "root directory" '/' "$lp_path"
+  assertContains "root directory formatting" "$lp_path_format" '{format}/'
+
+  LP_ENABLE_HYPERLINKS=1
+
+  local url="https://test.io/"
+  local label="liquid link"
+  local expected_link="$_LP_OPEN_ESC"$'\E]8;;'"${url}"$'\E'"${_LP_BACKSLASH}${_LP_CLOSE_ESC}${label}${_LP_OPEN_ESC}"$'\E]8;;\E'"${_LP_BACKSLASH}$_LP_CLOSE_ESC"
+  _lp_create_link "$url" "$label"
+  assertEquals "typical OSC-8 escape sequence" "$expected_link" "$lp_link"
+
+  url="https://test&eacute;.io/?var=1&dbg=2"
+  label="liquid\tlink"
+  expected_link="$_LP_OPEN_ESC"$'\E]8;;'"${url}"$'\E'"${_LP_BACKSLASH}${_LP_CLOSE_ESC}${label}${_LP_OPEN_ESC}"$'\E]8;;\E'"${_LP_BACKSLASH}$_LP_CLOSE_ESC"
+  _lp_create_link "$url" "$label"
+  assertEquals "typical OSC-8 escape sequence with complex text" "$expected_link" "$lp_link"
+
+  local pathword="home"
+  typeset PWD="/home/nojhan"
```

Comment:
Why mix `local` and `typeset`? In Bash and Zsh they mean the same thing.

The rest of the tests use `typeset` (because when I wrote them I was hoping we could get to Ksh support, but I gave that up, and Ksh does not support `local`), so just use that.

Reply:
Just copy/pasted without thinking about it. Which one do you prefer?

Reply:
I'd say `typeset`, to be consistent in the tests.